### PR TITLE
openfoam: disable FPE handling for Fujitsu compiler

### PR DESCRIPTION
**Pull request by olesenm**
closes #21336

- as outlined in merge-request #21336 some clang compilers
  can trigger erroneous floating point exceptions.

  OpenFOAM normally traps FPE, but disable this in the etc/controlDict
  for specific compilers:

      change "trapFpe digit;" -> "trapFpe 0;"

### CODE REVIEW

**Comment by olesenm** [comment]
@tldahlgren and @ShunIto-FJ

This would be my preferred solution. Making the change directly in the etc/controlDict will robuster. It won't matter if someone loads the module env or sources the bashrc directly.

**Comment by ShunIto-FJ** [comment]
@olesenm I agree with controlling the functionality of OpenFOAM's floating point exception traps with trapFpe in etc/controlDict instead of environment variables.
Notice that the environment variable FOAM_SIGFPE is set to false in setup_build_environment.
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/openfoam/package.py#L411

**Comment by t-karatsu** [comment]
Sorry for the late review.
I agree with @olesenm 's fix, but if it works, as @ShunIto-FJ says, the fix of https://github.com/spack/spack/pull/18601 may not be needed.

@ShunIto-FJ Please let us know the result of the operation check here.
@olesenm Could you please consider deleting the process of https://github.com/spack/spack/pull/18601 depending on  his result of the operation check?

**Comment by olesenm** [comment]
@alalazo - FYI might have crossed wires with respect to #18601

@t-karatsu - travelling at the moment, but will rebase to include a revert of #18601 after your confirmation

**Comment by olesenm** [comment]
> @olesenm I agree with controlling the functionality of OpenFOAM's floating point exception traps with trapFpe in etc/controlDict instead of environment variables.
> Notice that the environment variable FOAM_SIGFPE is set to false in setup_build_environment.
> https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/openfoam/package.py#L411

Would prefer to have it all controlled by the etc/controlDict. Setting the env variable false would indeed then be redundant, which is why it should be removed - it makes it appear to have more meaning than it should (and makes a messier environment)

**Comment by ShunIto-FJ** [comment]
@olesenm We will check the operation, so please wait for a while.
If I get the result, I will write it here.

**Comment by ShunIto-FJ** [comment]
@olesenm I used package.py in your pull request ( #21769 ) to check the operation of OpenFOAM v1812 and confirmed that it can be installed and executed without any problem.
In addition, I have confirmed that it can be installed and executed without any problem even if the correction included in the pull request of #18601 is reverted.

**Comment by alalazo** [comment]
@ShunIto-FJ @olesenm How should we proceed here? Merge #21769 and revert #18601 ?

**Comment by olesenm** [comment]
> @ShunIto-FJ @olesenm How should we proceed here? Merge #21769 and revert #18601 ?

@alalazo Missed the earlier notification, will force update tomorrow and tag everyone.

**Comment by olesenm** [comment]
@alalazo @t-karatsu @ShunIto-FJ  - ready to merge. Closes #21336

File: var/spack/repos/builtin/packages/openfoam/package.py [hunk 1]
```diff
@@ -405,10 +405,7 @@ def setup_minimal_environment(self, env):
     def setup_build_environment(self, env):
         """Sets the build environment (prior to unpacking the sources).
         """
-        # Avoid the exception that occurs at runtime
-        # when building with the Fujitsu compiler.
-        if self.spec.satisfies('%fj'):
-            env.set('FOAM_SIGFPE', 'false')
+        pass
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/openfoam/package.py:408]: Can we remove this function completely?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor issue. Otherwise LGTM.

File: var/spack/repos/builtin/packages/openfoam/package.py (hunk 1 above)
Comment (olesenm) [reply var/spack/repos/builtin/packages/openfoam/package.py:408]: maybe, but might still need to revive some hooks there in the future.
If you want to remove it, go ahead but my computer is shutdown for a while.

**Review by t-karatsu — APPROVED** [review]
Thanks @olesenm LGTM.

**Comment by olesenm** [comment]
@alalazo or @adamjstewart up for a merge?

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]
Not a fan of leaving stubs in, but we can remove it later.

**Comment by olesenm** [comment]
> Not a fan of leaving stubs in, but we can remove it later.

Okay, this will be a good next task I suppose.

**Pull request merged**